swagger_server/controllers/distance_controller.py

```diff
@@ -18,16 +18,20 @@ def samples_id_nearest_leaf_node_get(id):  # noqa: E501
     """
 
     try:
-        node = SampleNode.nodes.get(name=id)
-        leaf = node.lineage.get()
-        rel = node.lineage.relationship(leaf)
+        result = db.Neo4jDatabase.get().query(f'MATCH (n:SampleNode)-[r:LINEAGE]->(m:LineageNode) WHERE n.name="{id}" RETURN '
+                                         f'n,r,m').values()
```

Comment:
You can return only the interested properties from the query: https://neo4j.com/docs/cypher-manual/4.0/clauses/return/

This is to improve the performance. But in the future if the performance improvement is insignificant compared to the code reusability, then you can make it more generalised.